File: TransCoder_tokenized_test_set_functions/cpp/LEXICOGRAPHICALLY_PREVIOUS_PERMUTATION_IN_C.cpp
bool prevPermutation ( string & str ) {
  int n = str . length ( ) - 1;
  int i = n;
  while ( i > 0 && str [ i - 1 ] <= str [ i ] ) i --;
  if ( i <= 0 ) return false;
  int j = i - 1;
  while ( j + 1 <= n && str [ j + 1 ] <= str [ i - 1 ] ) j ++;
  swap ( str [ i - 1 ], str [ j ] );
  reverse ( str . begin ( ) + i, str . end ( ) );
  return true;
}

File: TransCoder_tokenized_test_set_functions/java/LEXICOGRAPHICALLY_PREVIOUS_PERMUTATION_IN_C.java
static boolean prevPermutation ( char [ ] str ) {
  int n = str . length - 1;
  int i = n;
  while ( i > 0 && str [ i - 1 ] <= str [ i ] ) {
    i --;
  }
  if ( i <= 0 ) {
    return false;
  }
  int j = i - 1;
  while ( j + 1 <= n && str [ j + 1 ] <= str [ i - 1 ] ) {
    j ++;
  }
  swap ( str, i - 1, j );
  StringBuilder sb = new StringBuilder ( String . valueOf ( str ) );
  sb . reverse ( );
  str = sb . toString ( ) . toCharArray ( );
  return true;
}